src/tool/csv2yaml.cpp

```diff
@@ -3082,12 +3082,25 @@ static bool itemdb_read_randomopt_group(char* str[], int columns, int current) {
 		return false;
 	}
 
-	uint16 id = static_cast<uint16>(rand_opt_group.size() + 1);
-	s_random_opt_group_csv *group = util::umap_find(rand_opt_group, id);
-	s_random_opt_group_csv group_entry;
+	s_random_opt_group group_entry;
```

Comment:
Just to note, this is the wrong way to go about this fix here. The csv2yaml should not be touched, but instead we use the yamlupgrade when drastic changes are needed to be made to the YAML structure. This way the content does not have to be converted from TXT to YAML again, simply just running yamlupgrade will convert the current YAML to the new YAML structure.

Reply:
Will look into yamlupgrade.
But shouldn't csv2yaml provide yaml files with the current format, instead of having to additionally run yamlupgrade (when not already having the yaml version)?

Reply:
Sorry, thought I replied before.

So let me take back some of what I stated before. To your question, yes we still want to maintain csv2yaml with the "latest" format but since some databases are conversions from csv and then we add features that weren't part of it, updating csv2yaml is moot since those pieces weren't in the old format.

In your case here you're adjusting prior content so it's ok!